src/core.ts

```diff
@@ -60,11 +62,14 @@ export function teardown() {
   }
 
   delete document[ClarityAttribute];
+  if (compressionWorker) {
+    compressionWorker.terminate();
```

Comment:
>terminate [](start = 22, length = 9)

Will this lose work in progress in the worker if we haven't gotten all messages back yet?

Reply:
Yes, 'terminate' kills the worker immediately. There might be messages that were already posted by the worker, but were not received by main thread yet, but we'll discard those based on the unloaded state.

---
In reply to: [137239598](https://github.com/Microsoft/clarity-js/pull/28#discussion_r137239598) [](ancestors = 137239598)

Reply:
Can you clarify what you mean "discard those based on unloaded state". Maybe put the answer in a comment here. 

---
In reply to: [137352823](https://github.com/Microsoft/clarity-js/pull/28#discussion_r137352823) [](ancestors = 137352823,137239598)

Reply:
'onWorkerMessage' message handler function starts with 'if (state !== State.Unloaded)', so all events that are delivered after teardown will be ignored.

---
In reply to: [137421841](https://github.com/Microsoft/clarity-js/pull/28#discussion_r137421841) [](ancestors = 137421841,137352823,137239598)

Reply:
Okay.

---
In reply to: [137422665](https://github.com/Microsoft/clarity-js/pull/28#discussion_r137422665) [](ancestors = 137422665,137421841,137352823,137239598)